tests/integration/test_gates_integration.py

```diff
@@ -513,3 +516,108 @@ def test_quantum_multi_hadamard_partial_measure():
     assert result_1["1000"] == result_1["1001"] == result_1["1010"] == result_1["1011"] == 0
     assert result_1["1100"] == result_1["1101"] == result_1["1110"] == result_1["1111"] == 0
     assert result_1["0100"] == result_1["0101"] == result_1["0110"] == result_1["0111"] == 0
+
+
+def test_grover_search_for_6():
+    grover = Circuit()
+    grover.add(Qubits(3))
+    grover.add(H(range(3)))
+
+    for i in range(2):
+        grover.add(H(2))
+        grover.add(PauliX(0))
+        grover.add(CCNOT(0, 1, 2))
+        grover.add(PauliX(0))
+        grover.add(H(2))
+
+        grover.add(H(range(3)))
+        grover.add(PauliX(range(3)))
+        grover.add(H(2))
+        grover.add(CCNOT(0, 1, 2))
+        grover.add(H(2))
+        grover.add(PauliX(range(3)))
+        grover.add(H(range(3)))
+
+    grover.add(Measure(0, 1, 2))
+    job = grover.run(1024, QiskitProvider(backend="qasm_simulator"))
+    result = job.result
+
+    assert result[6] > 900
+    assert result[0] < 25
+    assert result[1] < 25
+    assert result[2] < 25
+    assert result[3] < 25
+    assert result[4] < 25
+    assert result[5] < 25
+
+
+@pytest.mark.xfail(reason="Don't have capability to multiple qbit classes in one QC and index between them")
+def test_mlt_qbit_mlt_cbit1():
+    qbits1 = Qbits(1)
+    qbits2 = Qbits(1)
+    circ = QC()
+    circ.add(qbits1)
+    circ.add(qbits2)
+    circ.add(H(qbits1))
+    circ.add(X(qbits2[2]))
+    circ.add(Measure(qbits1[0], qbits2[1], Cbits[2]))
+
+    result = X.run(200).result
+    assert result["01"] > 180
+    assert result["11"] > 180
+
+
+def test_mlt_qbit_mlt_cbit3():
+    qbits1 = Qbits(1)
+    qbits2 = Qbits(2)
+    circ = QC()
+    circ.add(qbits1)
+    circ.add(qbits2)
+    circ.add(H(0))
+    circ.add(X(1))
+    circ.add(CH(1, 2))
+    circ.add(Measure(0, 1, output_bits=[0, 1]))
+
+    result = circ.run(1000).result
+    print(result)
+    assert result["10"] > 450
+    assert result["11"] > 450
+
+
+def test_mlt_qbit_mlt_cbit4():
+    qbits1 = Qbits(1)
+    qbits2 = Qbits(3)
+    circ = QC()
+    circ.add(qbits1)
+    circ.add(qbits2)
+    circ.add(Cbits(3))
+    circ.add(H(0))
+    circ.add(X(1))
+    circ.add(CH(1, 2))
+    circ.add(Measure(0, 1, output_bits=[0, 1]))
```

Comment:
So this works because we are mapping the same relative indexes to eachother.

This breaks if we try to do any other mapping correct?